spec/support/sdam_monitoring/replica_set_with_removal.yml

```diff
@@ -0,0 +1,88 @@
+description: "Monitoring a replica set with non member"
+uri: "mongodb://a,b/?replicaSet=rs"
+phases:
+  -
+    responses:
+      "a:27017":
+        ok: 1
+        ismaster: true
+        setName: "rs"
+        setVersion: 1.0
+        hosts: [ "a:27017" ]
+        minWireVersion: 0
+        maxWireVersion: 4
+      "b:27017":
+        ok: 1
+        ismaster: true
+    outcome:
+      events:
+        -
+          topology_opening_event:
+            topologyId: "42"
+        -
+          server_opening_event:
+            topologyId: "42"
+            address: "a:27017"
+        -
+          server_opening_event:
+            topologyId: "42"
+            address: "b:27017"
+        -
+          server_description_changed_event:
+            topologyId: "42"
+            address: "a:27017"
+            previousDescription:
+              address: "127.0.0.1:27017"
+              arbiters: []
+              electionId: null
+              error: null
+              hosts: []
+              maxWireVersion: 0
+              me: null
+              minWireVersion: 0
+              passives: []
+              primary: null
+              setName: null
+              setVersion: null
+              tags: []
+              type: "Unknown"
+            newDescription:
+              address: "a:27017"
+              arbiters: []
+              electionId: null
+              error: null
+              hosts: [ "a:27017" ]
+              maxWireVersion: 4
+              me: "a:27017"
+              minWireVersion: 0
+              passives: []
+              primary: true
+              setName: "rs"
+              setVersion: 1.0
+              tags: []
+              type: "RSPrimary"
+        -
+          server_closed_event:
+            topologyId: "42"
+            address: "b:27017"
+        -
+          topology_description_changed_event:
+            topologyId: "42"
+            previousDescription:
+              type: "Unknown"
+              setName: null
+              maxSetVersion: null
```

Comment:
We should consider removing some of these fields.  While it's true they are all part of the TopologyDescription as defined in the specification, most are not useful to clients and should be considered implementation details of the SDAM implementation.  The only ones generally useful to clients are type and setName and servers. 

As for servers, the spec says that it's a list of ServerDescription, not a list of server address strings.  And that's necessary to actually implement the hasReadable/WritableServer methods.  To avoid repetition in the YAML, you can use node anchors (http://yaml.org/spec/1.2/spec.html#id2785586)